### PR TITLE
Rename some FFI constructors

### PR DESCRIPTION
**Pull request by robertbastian**
Renames
* `create_from_*` -> `from_*`, e.g. `create_from_string` -> `from_string`
* `create_*` -> `*` where the constructor doesn't take arguments, e.g. `create_und`, -> `und`

These names were already used as named constructors, so this basically aligns C and C++ with that.

### CODE REVIEW

File: ffi/capi/src/collator.rs [hunk 1]
```diff
@@ -31,7 +31,7 @@ pub mod ffi {
     // `ResolvedCollatorOptions` makes more sense as English.
     #[diplomat::rust_link(icu::collator::ResolvedCollatorOptions, Struct)]
     #[diplomat::out]
-    #[diplomat::attr(any(dart, js), rename = "ResolvedCollatorOptions")]
+    #[diplomat::attr(any(dart, js), rename = "CollatorResolvedOptions")]
```
Comment (Manishearth) [inline ffi/capi/src/collator.rs:34]: thought: (followup ok) ought this be `supports = non_exhaustive_structs`?

and below

fine with this happening in a different PR

**Review by Manishearth — APPROVED** [review]

**Review by Manishearth — DISMISSED** [review]
nice

**Review by Manishearth — APPROVED** [review]

**Comment by sffc** [comment]
I think we called them all "create" to make it clear which functions were constructors, but that was before we had a constructor annotation.

**Comment by robertbastian** [comment]
Yes

**Pull request merged**